src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -764,29 +764,39 @@ static function (ChildDefinition $definition, AsPeriodicTask|AsCronTask $attribu
             );
         }
 
-        $container->registerForAutoconfiguration(CompilerPassInterface::class)
-            ->addExcludeTag('container.excluded.compiler_pass');
-        $container->registerForAutoconfiguration(TestCase::class)
-            ->addExcludeTag('container.excluded.test_case');
-        $container->registerAttributeForAutoconfiguration(AsMessage::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.messenger.message');
-        });
-        $container->registerAttributeForAutoconfiguration(Entity::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.doctrine.entity');
-        });
-        $container->registerAttributeForAutoconfiguration(Embeddable::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.doctrine.embeddable');
-        });
-        $container->registerAttributeForAutoconfiguration(MappedSuperclass::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.doctrine.mapped_superclass');
-        });
-
-        $container->registerAttributeForAutoconfiguration(JsonStreamable::class, static function (ChildDefinition $definition, JsonStreamable $attribute) {
-            $definition->addExcludeTag('json_streamer.streamable', [
-                'object' => $attribute->asObject,
-                'list' => $attribute->asList,
-            ]);
-        });
+        if (method_exists(Definition::class, 'addExcludeTag')) {
+            $container->registerForAutoconfiguration(CompilerPassInterface::class)
+                ->addExcludeTag('container.excluded.compiler_pass');
+            $container->registerForAutoconfiguration(TestCase::class)
+                ->addExcludeTag('container.excluded.test_case');
+            $container->registerAttributeForAutoconfiguration(AsMessage::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.messenger.message');
+            });
+            $container->registerAttributeForAutoconfiguration(Entity::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.entity');
+            });
+            $container->registerAttributeForAutoconfiguration(Embeddable::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.embeddable');
+            });
+            $container->registerAttributeForAutoconfiguration(MappedSuperclass::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.mapped_superclass');
+            });
```

Comment:
This needs to be declared even if `symfony/dependency-injection < 7.3` is used. Because the Flex recipe for `symfony/framework-bundle 7.3` relies on it.

The declaration from the `DoctrineBundle` can be copied: And remove the tag name that is not used for Doctrine classes; as done in DoctrineBundle. https://github.com/doctrine/DoctrineBundle/blob/a91383d6d8788f8c63dbf9e6d51677d5d6fac093/src/DependencyInjection/DoctrineExtension.php#L650-L658

Reply:
I wonder if 32c994c81e2bfd4c9a05fc40bb76d5752cdc5ff7 then isn't a better solution.